src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -417,46 +405,36 @@ static bool IsAsyncMethod(MethodInfo method)
                 // No default either. Leave it empty.
                 return null;
             };
-
-            string? description = parameter.GetCustomAttribute<DescriptionAttribute>(inherit: true)?.Description;
-            return new AIFunctionParameterMetadata(parameter.Name)
-            {
-                Description = description,
-                HasDefaultValue = parameter.HasDefaultValue,
-                DefaultValue = parameter.HasDefaultValue ? parameter.DefaultValue : null,
-                IsRequired = !parameter.IsOptional,
-                ParameterType = parameter.ParameterType,
-            };
         }
 
         /// <summary>
         /// Gets a delegate for handling the result value of a method, converting it into the <see cref="Task{FunctionResult}"/> to return from the invocation.
         /// </summary>
-        private static Type GetReturnMarshaller(MethodInfo method, out Func<object?, ValueTask<object?>> marshaller)
+        private static Func<object?, ValueTask<object?>> GetReturnMarshaller(MethodInfo method, out Type returnType)
```

Comment:
Curious what motivated swapping the return and out?

Reply:
The name of the method suggests that the return type should be the marshaller. Results in cleaner code in the method body I think.